src/index.ts

```diff
@@ -12,6 +12,72 @@ export interface AccessControlOptions {
   preflightContinue?: boolean
 }
 
+function isIterable(obj: unknown): obj is Iterable<unknown> {
+  return typeof obj[Symbol.iterator] === 'function'
+}
+
+function getOriginHeaderHandler(origin: unknown): (req: Request, res: Response) => void {
+  function fail() {
+    throw new TypeError('No other objects allowed. Allowed types is array of strings or RegExp')
+  }
+
+  if (typeof origin === 'boolean' && origin === true) {
+    return function (_, res) {
+      res.setHeader('Access-Control-Allow-Origin', '*')
+    }
+  }
+
+  if (typeof origin === 'boolean' && origin === false) {
```

Comment:
you can do this instead:

```js
if (typeof origin === 'boolean') {
  return origin ? () => undefined : (_ res) => {
    res.setHeader('Access-Control-Allow-Origin', '*')
  }
}

Reply:
You can, and it is more concise, but I think it sacrifices too much readability. 

I will make the change regardless if it is your preference.

Reply:
it doesn't make it any less readable. anyone who knows how Boolean statements work will understand whats going on

Reply:
yes, that's my point - you have to be familiar with JavaScript's ternary operator to be able to read it. 

`if` is language-agnostic; almost any programmer can read it.

Reply:
you can rewrite it with an if as well.But making 3 separate ifs makes no sense to me

Reply:
It doesn't hurt performance because the `getOriginHandler` method is only evaluated once, when the middleware is instantiated.

If someone cares about the difference between a ternary operator's evaluation and two `if`s at startup time, they shouldn't be using JavaScript! 😅